src/app/stores/officeHourStore.ts

```diff
@@ -0,0 +1,51 @@
+import { createStore } from "zustand";
+import { IdentifiableCourse, IdentifiableQuestions } from "@interfaces/type";
+import { doc, onSnapshot, collection } from "firebase/firestore";
+import { db } from "@project/firebase";
+import { courseConverter, questionConverter } from "@services/firestore";
+
+interface OfficeHourState {
+  course: IdentifiableCourse | null;
```

Comment:
What's the downside if we were to initialize this to an empty `IdentifiableCourse`? I'm not a fan of working with nulls; alternatively, it would be better if we ensure that the store can be used if and only if it has been initialized.